shiny/ui/_input_action_button.py

```diff
@@ -64,7 +64,15 @@ def input_action_button(
     return tags.button(
         {"class": "btn btn-default action-button", "style": css(width=width)},
         icon,
-        None if icon is None else " ",
+        # The separator element helps us distinguish between the icon and label
+        # when dynamically updating the button/link (see rstudio/shiny#4242)
+        tags.span(class_="shiny-icon-separator") if icon else None,
+        # Adds a space between the icon and label
+        # TODO: this space gets removed when icon/label are updated dynamically,
+        # which is not ideal. The 'right' way to do this would be to either
+        # add a CSS class to the separator element, or wrap both the icon and
+        # label in a container element.
+        " " if icon and label else None,
```

Comment:
This obviously isn't ideal, and long-term this isn't the approach to spacing that we'd want, but I'm thinking I'll just keep to retain the current behavior. 

Kinda makes me wish we didn't decide to revert https://github.com/rstudio/shiny/pull/4248

Reply:
Ugh, the old JS logic would (unconditionally, for both button and links!!) when updates to either icon or label occur

https://github.com/rstudio/shiny/blob/33dc41c/srcts/src/bindings/input/actionbutton.ts#L73

Which is a big bummer because, unless we add the space in the server-side update, the space gets removed 😭